src/CMakeLists.txt

```diff
@@ -1,61 +1,95 @@
+# find required packages for base and app
+set(LibtorrentRasterbar_DIR "" CACHE PATH "Path to LibtorrentRasterbar CMake configuration files")
```

Comment:
It confuses me? Why it's really needed? And if it's needed why only for Libtorrent?

Reply:
@glassez It was needed to fix a problem with reconfiguration with `ccmake` mentioned a few comments up: https://github.com/qbittorrent/qBittorrent/pull/12746#issuecomment-652453467

Reply:
>But it looks like simply adding the type makes it work: cmake -DLibtorrentRasterbar_DIR:PATH=~/build/libtorrent/LibtorrentRasterbar ~/qbittorrent_sources. I investigated the generated CMake cache in both cases, and it seems the problem is the CMake GUIs don't expose UNINITIALIZED variables for the user to change.
I pushed a new commit to make it so that CMake always knows LibtorrentRasterbar_DIR is a PATH, by setting it as a CACHE variable. Now, even if you don't specify the type in the command line, the variable will always show up and be configurable in a GUI such as ccmake.

Doesn't seem like a problem that should be fixed in this way. You just need to set variable with a type. If it's unexpected/undocumented behavior it's definitely CMake issue.

Reply:
@glassez 
> You just need to set variable with a type.

Exactly, that's what this line does - ensures the variable has a type. It initializes the variable in the cache with no value (empty string), but with `PATH` type. Nothing else is affected by this, and this allows `ccmake` to work correctly.

From the CMake docs (https://cmake.org/cmake/help/v3.16/command/set.html):

>  Since cache entries are meant to provide user-settable values this does not overwrite existing cache entries by default.

(...)

> It is possible for the cache entry to exist prior to the call but have no type set if it was created on the `cmake(1)` command line by a user through the `-D<var>=<value>` option without specifying a type. In this case the set command will add the type. Furthermore, if the `<type>` is `PATH` or `FILEPATH` and the `<value>` provided on the command line is a relative path, then the set command will treat the path as relative to the current working directory and convert it to an absolute path.

Reply:
Still doesn't seem good for me.
I can't understand what you're doing with this. Is this the intended way? Or is this a mistake (of CMake) you're trying to workaround? Or is it a "feature" when you cache a variable that wasn't originally intended to be cached?
Anyway, it looks strange to affect a variable that isn't even used directly in your script. And then it turns out that in general case you should do the same with all possible variables that can ever be set via the command line... it looks terrible.
Is there no other way?

Reply:
@glassez 

To be honest, I also get the feeling that this is simply a workaround for cmake GUIs like (`ccmake`, in particular). This is a workaround for the case where the user initially specifies a variable in the command line without a type (I would says it's uncommon for CMake users to specify variables _with_ a type in the command line invocation) and then later would like to change the variable's value with `ccmake`. It will not be possible, because `ccmake` does not allow you to change variables that don't have a specified type.

I have found that some tools that invoke CMake, such as the CMake Tools extension for VS Code, add a variable type to all command-line entries, side-stepping this problem entirely.

As for, "why have this `set(CACHE)` call for `LibtorrentRasterbar_DIR` and not others?", I would say this is the one variable that's not used directly that will be changed most often, and also this way the workflow of a contributor isn't broken. But if we go down this route, then yes, it might become justifiable to do this for other variables as well. But that would be done in another PR.
I guess it comes down to whether or not you want to leave this workaround in to better support GUI users. Unless you know of a better way to achieve the same thing.